app/controllers/homepage_controller.rb

```diff
@@ -91,6 +91,8 @@ def index
                  listing_shape_menu_enabled: listing_shape_menu_enabled }
       }
     end
+
+    @main_search = MarketplaceService::API::Marketplaces.search_settings(community_id: @current_community.id).data[:main_search]
```

Comment:
Yeah, I was kind of thinking between these options. 
I thought, that since homepage-index is a huge function and new additions have not take that step there might be a reason / plan to make changes on one huge swipe when the new browse view is coming.